include/boost/math/cstdfloat/cstdfloat_limits.hpp

```diff
@@ -24,7 +24,7 @@
 #pragma GCC system_header
 #endif
 
-  #if defined(BOOST_CSTDFLOAT_HAS_INTERNAL_FLOAT128_T) && defined(BOOST_MATH_USE_FLOAT128) && !defined(BOOST_CSTDFLOAT_NO_LIBQUADMATH_SUPPORT)
+  #if defined(BOOST_CSTDFLOAT_HAS_INTERNAL_FLOAT128_T) && defined(BOOST_MATH_USE_FLOAT128) && !defined(BOOST_CSTDFLOAT_NO_LIBQUADMATH_SUPPORT) && (!defined(__GNUC__) || (defined(__GNUC__) && __GNUC__ < 14))
```

Comment:
This check seems wrong. The new `numeric_limits<__float128>` specialization will be enabled when using new libstdc++ headers with Clang, which does not define `__GNUC__` to 14 (it can be told to use any value, but by default it defines it to 4).